compiler/src/dotty/tools/dotc/semanticdb/Main.scala

```diff
@@ -0,0 +1,24 @@
+package dotty.tools.dotc.semanticdb
+
+import java.nio.file.Files
+
+import dotty.tools.dotc
+import dotty.tools.dotc.core.Contexts._
+
+/** Main class of the `dotc -gen-semantic-db`.
+ *
+ * @author Nicolas Stucki
+ */
+object Main extends dotc.Driver {
+  override protected def newCompiler(implicit ctx: Context): dotc.Compiler = {
+    assert(ctx.settings.fromTasty.value)
+    new TASTyFromClass
+  }
+
+  override def setup(args0: Array[String], rootCtx: Context): (List[String], Context) = {
+    var args = args0.filter(a => a != "-gen-semantic-db")
+    if (!args.contains("-from-tasty")) args = "-from-tasty" +: args
+    if (args.contains("-d")) args = "-color:never" +: args
```

Comment:
?

Reply:
Copy pasted from the decompiler. Will remove it.